packages/teams-js/src/public/dialog/adaptiveCard/adaptiveCard.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { sendMessageToParent } from '../../../internal/communication';
+import { dialogTelemetryVersionNumber, getDialogInfoFromAdaptiveCardDialogInfo } from '../../../internal/dialogHelpers';
+import { ensureInitialized } from '../../../internal/internalAPIs';
+import { ApiName, getApiVersionTag } from '../../../internal/telemetry';
+import { isHostAdaptiveCardSchemaVersionUnsupported } from '../../../internal/utils';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../../constants';
+import { AdaptiveCardDialogInfo, DialogInfo } from '../../interfaces';
+import { runtime } from '../../runtime';
+import { DialogSubmitHandler } from '../dialog';
+import * as url from '../url/url';
+import * as bot from './bot';
+
+/**
+ * Subcapability for interacting with adaptive card dialogs
+ * @beta
+ */
+/**
+ * Allows app to open an adaptive card based dialog.
+ *
+ * @remarks
+ * This function cannot be called from inside of a dialog
+ *
+ * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module {@link AdaptiveCardDialogInfo}.
+ * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode url.submit} function or when the user closes the dialog.
```

Comment:
I think this comment is actually wrong, and that fixing it will let you remove the reference to the url file.

For adaptive card dialogs, this handler is not triggered by calls to the submit function because that function can only be called from url dialogs which are essentially Teams apps running inside of a dialog shell. Instead, for adaptive card dialogs the equivalent is when the adaptive card fires an [Action.Submit](https://adaptivecards.io/explorer/Action.Submit.html). If we update this comment to be something like this:

```suggestion
 * @param submitHandler - Handler that triggers when a dialog fires an [Action.Submit](https://adaptivecards.io/explorer/Action.Submit.html) or when the user closes the dialog.
```

This will also make sure we don't unintentionally use any url-only logic in the adaptive card functionality

Reply:
Fixed!